### PR TITLE
Task/dr 396 slack notifiy ghw new

### PR DESCRIPTION
**Pull request by stuartreid**
## Description

When a github action is triggered either in the dev or test environment on certain pipelines a slack notification will be sent to the appropriate channel if the pipeline fails. No notification will be sent if the pipeline is successful.

**Note**
https://nhsd-jira.digital.nhs.uk/browse/DR-396

## Context

DR-396 - Add slack integration for github action workflow notification

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/workflows/pipeline-deploy-bootstrap.yaml [hunk 1]
```diff
@@ -18,3 +18,13 @@ jobs:
       with:
         ENV: test
       secrets: inherit
+    slack-notifications:
+      needs: [
+        deploy-bootstrap-dos-dev,
+      ]
```
Comment (timrickwood) [inline .github/workflows/pipeline-deploy-bootstrap.yaml:24]: I think we should add deploy-bootstrap-dos-test to the needs list. Both the deploy dev and deploy test jobs will run in parallel and we want to know if either of them fails

File: .github/workflows/slack-notifications.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      with:
+        payload: |
+            {
+            "text": "GitHub Action build result: ${{ env.WORKFLOW_CONCLUSION }}\nhttps://github.com/NHSDigital/uec-dos-service-management/actions/runs/${{ github.run_id }} \nPushed by: @${{ github.actor }}",
+            "blocks": [
+                {
+                "type": "section",
+                "text": {
+                    "type": "mrkdwn",
+                    "text": "GitHub Action build result: ${{ env.WORKFLOW_CONCLUSION }}\nhttps://github.com/NHSDigital/uec-dos-service-management/actions/runs/${{ github.run_id }}\nPushed by: @${{ github.actor }}"
```
Comment (timrickwood) [inline .github/workflows/slack-notifications.yaml:45]: Not for today but I will raise a ticket to cover it - I think we could make this workflow more portable between repos if we maybe parameterise the link to the failed workflow.

File: .github/workflows/pipeline-deploy-bootstrap.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (stuartreid) [reply .github/workflows/pipeline-deploy-bootstrap.yaml:25]: done

File: .github/workflows/pipeline-deploy-app-test.yaml [hunk 3]
```diff
@@ -123,3 +123,17 @@ jobs:
         github_environment: test
         DEPLOYMENT_WORKSPACE: ${{ needs.derive-deployment-workspace.outputs.deployment_workspace }}
       secrets: inherit
+  slack-notifications:
```
[the remark this answers is not in the document]
Comment (stuartreid) [reply .github/workflows/pipeline-deploy-app-test.yaml]: removed as suggested

**Review by timrickwood — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**